<font size = 2> <B> PHILIP AUERBACH <BR>UNI: PJA2113 </B></font>

<font size = 2> <B> Gender Beliefs and Religion </B></font><br> <B> Lab #3 </B> 


<b> Project Setup: </b>

In [94]:
# Importing packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import statsmodels.formula.api as smf 
import scipy.stats as stats

# Grabbing Data
url = 'https://www.qogdata.pol.gu.se/data/qog_std_ts_jan25.csv'
df = pd.read_csv(url)

# Visualize 5 first rows of data
df.head()

/var/folders/yk/ljnwx5m95179j3_74vk6g29r0000gn/T/ipykernel_1091/265183431.py:12: DtypeWarning: Columns (1362,1363) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(url)


cname_qog        cname  year  ccodecow ccodealp ccodealp_year  ccode_qog  \
0  Afghanistan  Afghanistan  1946     700.0      AFG         AFG46          4   
1  Afghanistan  Afghanistan  1947     700.0      AFG         AFG47          4   
2  Afghanistan  Afghanistan  1948     700.0      AFG         AFG48          4   
3  Afghanistan  Afghanistan  1949     700.0      AFG         AFG49          4   
4  Afghanistan  Afghanistan  1950     700.0      AFG         AFG50          4   

         cname_year  ccode  aid_cpnc  ...  yri_fem40  yri_fem4160  yri_fem61  \
0  Afghanistan 1946    4.0       NaN  ...        NaN          NaN        NaN   
1  Afghanistan 1947    4.0       NaN  ...        NaN          NaN        NaN   
2  Afghanistan 1948    4.0       NaN  ...        NaN          NaN        NaN   
3  Afghanistan 1949    4.0       NaN  ...        NaN          NaN        NaN   
4  Afghanistan 1950    4.0       NaN  ...        NaN          NaN        NaN   

   yri_meanage  yri_medianage  yri_mp30  yri_mp35  yri_mp40  yri_mp4160  \
0          NaN            NaN       NaN       NaN       NaN         NaN   
1          NaN            NaN       NaN       NaN       NaN         NaN   
2          NaN            NaN       NaN       NaN       NaN         NaN   
3          NaN            NaN       NaN       NaN       NaN         NaN   
4          NaN            NaN       NaN       NaN       NaN         NaN   

   yri_mp61  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN  

[5 rows x 2009 columns]

<b> Question 4.1 </b> <br>
# **1-- Run a naive OLS regression on your time series data. 

Tell me how you expect your Xs to affect your Y and why.  

 Interpret your results.**


In the QoG file, I find a interesting Variable I wanted to explore more.    wvs_menpol 

For each of the following statements I read out, can you tell me how much you agree with each.
"Men make better political leaders than women do."


In [95]:
# Variable 1, Men make better political leaders than women do, wvs_menpol 
'''
Do you agree strongly, agree, disagree, or disagree strongly?

1 Strongly disagree
2 Disagree
3 Agree
4 Strongly agree

From QoG Codebook

'''

df['wvs_menpol'].describe()

count    302.000000
mean       2.453618
std        0.468024
min        1.306160
25%        2.103288
50%        2.459855
75%        2.763895
max        3.640806
Name: wvs_menpol, dtype: float64

In [96]:
#Variable 2, Official Religion measurement   biu_offrel

"""
Official Religion measures whether the government has an established religion. For a religion to be established there must be a constitutional clause, a law, or the equivalent explicitly stating that a specific religion or specific religions are the official religions of that state. This variable is coded on the following scale:

0 The state has no official religion
1 The state has multiple established religions
2 The state has one established religion

""" 
df['biu_offrel'].describe()

count    4424.000000
mean        0.442360
std         0.826765
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max         2.000000
Name: biu_offrel, dtype: float64

In [97]:
# Variable 3, Politicians who don’t believe in God are unfit for public office 
'''
1. Strongly disagree
2. Disagree
3. Neutral
4. Agree
5. Strongly Agree

From QoG Codebook

'''
df['wvs_godpol'].describe()

count    156.000000
mean       2.834254
std        0.775118
min        1.343166
25%        2.305337
50%        2.704683
75%        3.338695
max        4.762500
Name: wvs_godpol, dtype: float64

The dependant Variable is a fairly conservative statement. I expect to see countries that have higher beliefs in the independent variables statements to also have a higher belief in the depdent variable. 

In [98]:
gdp1 = smf.ols(formula = 'wvs_menpol ~ biu_offrel + wvs_godbel', data = df).fit()  
print (gdp1.summary())

                            OLS Regression Results                            
Dep. Variable:             wvs_menpol   R-squared:                       0.157
Model:                            OLS   Adj. R-squared:                  0.144
Method:                 Least Squares   F-statistic:                     12.38
Date:                Fri, 27 Jun 2025   Prob (F-statistic):           1.17e-05
Time:                        23:14:35   Log-Likelihood:                -67.086
No. Observations:                 136   AIC:                             140.2
Df Residuals:                     133   BIC:                             148.9
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.9176      0.173     11.069      0.0

For every one unit increase in belief that men make better politcal leaders, we see a 0.66 increase in belief that atheist politcians are unfit for public office. And we also see a small rise in religous countries. 

With a R^2 of 0.157 these two variables only let us predict 15.7% of all movement of of wvs_menpol. 

<b> Question 4.2 </b> <br>
Run a first differences regression on the same model in Question 1.
Interpret your results.
Do you draw a different conclusion than in Question 1? Explain.

In [99]:
!pip install linearmodels

from linearmodels.panel import FirstDifferenceOLS

In [111]:
columns = ['ccode', 'year', 'wvs_menpol', 'biu_offrel', 'wvs_godpol']
df1 = df[columns].dropna()

# Set the MultiIndex for panel data
df1 = df1.set_index(['ccode', 'year'])

df1 = df1.loc[~df1.index.duplicated(keep='first')]

# Define the dependent and independent variables
y = df1['wvs_menpol']
X = df1[['biu_offrel', 'wvs_godpol']]

# Fit the first-differenced panel data model
fdmodel = FirstDifferenceOLS(y, X)
results = fdmodel.fit(cov_type='clustered', cluster_entity=True)

print(results)


                     FirstDifferenceOLS Estimation Summary                      
Dep. Variable:             wvs_menpol   R-squared:                        0.0252
Estimator:         FirstDifferenceOLS   R-squared (Between):              0.0987
No. Observations:                   8   R-squared (Within):               0.1466
Date:                Fri, Jun 27 2025   R-squared (Overall):              0.0980
Time:                        23:17:51   Log-likelihood                    9.6212
Cov. Estimator:             Clustered                                           
                                        F-statistic:                      0.0774
Entities:                          62   P-value                           0.9264
Avg Obs:                       1.2903   Distribution:                     F(2,6)
Min Obs:                       1.0000                                           
Max Obs:                       2.0000   F-statistic (robust):          2.281e+16
                            

Running this data to see the relationship over time, leads to a different result. Here we have a much lower R^2 of 0.025